yt_dlp/extractor/floatplane.py

```diff
@@ -0,0 +1,268 @@
+
+            path_template = traverse_obj(stream, ('resource', 'uri', {str}))
+
+            def format_path(params):
+                path = path_template
+                for i, val in (params or {}).items():
+                    path = path.replace(f'{{qualityLevelParams.{i}}}', val)
+                return path
+
+            formats = []
+            for quality in traverse_obj(stream, ('resource', 'data', 'qualityLevels', ...)):
+                url = urljoin(stream['cdn'], format_path(traverse_obj(
+                    stream, ('resource', 'data', 'qualityLevelParams', quality['name']))))
+                formats.append({
+                    **traverse_obj(quality, {
+                        'format_id': 'name',
+                        'format_note': 'label',
+                        'width': ('width', {int}),
+                        'height': ('height', {int}),
+                    }),
+                    **parse_codecs(quality.get('codecs')),
+                    'url': url,
+                    'ext': determine_ext(url.partition('/chunk.m3u8')[0], 'mp4'),
+                })
+
+            items.append({
+                'id': media_id,
+                **traverse_obj(metadata, {
+                    'title': 'title',
+                    'duration': ('duration', {int_or_none}),
+                    'thumbnail': ('thumbnail', 'path'),
+                }),
+                'formats': formats,
+            })
+
+        uploader_url = format_field(traverse_obj(
+            post_data, ('creator', 'urlname')), template='https://www.floatplane.com/channel/%s/home', default=None)
+        channel_url = format_field(traverse_obj(
+            post_data, ('channel', 'urlname')), template=f'{uploader_url}/%s', default=None)
+
+        post_info = {
+            'id': post_id,
+            'display_id': post_id,
+            **traverse_obj(post_data, {
+                'title': 'title',
+                'description': ('text', {clean_html}),
+                'uploader': ('creator', 'title'),
+                'uploader_id': ('creator', 'id'),
+                'channel': ('channel', 'title'),
+                'channel_id': ('channel', 'id'),
+                'like_count': ('likes', {int_or_none}),
+                'dislike_count': ('dislikes', {int_or_none}),
+                'comment_count': ('comments', {int_or_none}),
+                'release_timestamp': ('releaseDate', {parse_iso8601}),
+                'thumbnail': ('thumbnail', 'path'),
+            }),
+            'uploader_url': uploader_url,
+            'channel_url': channel_url,
+            'availability': self._availability(needs_subscription=True),
+        }
+
+        if len(items) > 1:
+            return self.playlist_result(items, **post_info)
+
+        post_info.update(items[0])
+        return post_info
+
+
+class FloatplaneChannelIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:(?:www|beta)\.)?floatplane\.com/channel/(?P<id>[\w_]+)/home(?:/(?P<channel>[\w_]+))?'
```

Comment:
`_` is matched by `\w`
```suggestion
    _VALID_URL = r'https?://(?:(?:www|beta)\.)?floatplane\.com/channel/(?P<id>\w+)/home(?:/(?P<channel>\w+))?'
```

Reply:
That was meant to be a `-`. While there are no channels right now with a dash in their name, I think it's more future-proof to include.